### PR TITLE
[5.2] Allow Validator Interaction on ValidatesRequests

### PR DESCRIPTION
**Pull request by CupOfTea696**
Allow interaction with the Validator instance on ValidatesRequests by calling passing a callback as last parameter in `validate` or `validateWithBag`.

Example:

``` php
/**
 * Store a new game collector.
 *
 * @param  Request  $request
 * @return Response
 */
public function store(Request $request)
{
    $this->validate($request, [
        'email' => 'required|email',
        'games' => 'required|numeric',
    ], [], [], function($validator) {
        $validator->sometimes('reason', 'required|max:500', function($input) {
            return $input->games >= 100;
        });
    });

    // The game collector is valid, store in database...
}
```

### CODE REVIEW

File: src/Illuminate/Foundation/Validation/ValidatesRequests.php [hunk 1]
```diff
@@ -11,6 +11,13 @@
 trait ValidatesRequests
 {
     /**
+     * Conditional validation rules based on a Closure.
+     *
+     * @var array
+     */
+    protected $sometimes = [];
```
Comment (lucasmichot) [inline src/Illuminate/Foundation/Validation/ValidatesRequests.php]: This won't work if I already have a `$sometimes` attribute in the controller using `ValidatesRequests `

Comment (CupOfTea696) [inline src/Illuminate/Foundation/Validation/ValidatesRequests.php]: I could rename the variable to something more unique, but it's not a problem you can ever fully get rid of. I mean, this trait also sets `$validatesRequestErrorBag`.

**Comment by localdisk** [comment]
:+1:

**Comment by garygreen** [comment]
To be honest I think this could get a bit messy. The ValidatesRequest shouldn't be responsible for that kind of thing, what about if you wanted to add `->each` and stuff like that to the validator? I would much prefer to allow passing of a closure to $validate I don't like the whole inversion of control here.. I would MUCH rather have a closure supplied to the `validate` method that will be called before validation occurs. That will allow you to interact with the validator object and add `sometimes` `each` etc

```php
/**
 * Store a new game collector.
 *
 * @param  Request  $request
 * @return Response
 */
public function store(Request $request)
{
    $this->validate($request, [
        'email' => 'required|email',
        'games' => 'required|numeric',
    ], [], [], function($validator) {
    	$validator->sometimes('reason', 'required|max:500', function($input) {
        	return $input->games >= 100;
        });
    });

    // The game collector is valid, store in database...
}
```

**Comment by CupOfTea696** [comment]
@garygreen That's an excellent idea!

File: src/Illuminate/Foundation/Validation/ValidatesRequests.php [hunk 2]
```diff
@@ -11,6 +11,13 @@
 trait ValidatesRequests
 {
     /**
+     * Conditional validation rules based on a Closure.
+     *
+     * @var array
+     */
+    protected $validatesRequestSometimes = [];
```
Comment (garygreen) [inline src/Illuminate/Foundation/Validation/ValidatesRequests.php]: No longer need this

Comment (CupOfTea696) [inline src/Illuminate/Foundation/Validation/ValidatesRequests.php]: Whoops missed that one.

**Comment by garygreen** [comment]
Might want to squash those commits down as well.

**Comment by CupOfTea696** [comment]
Well, I tried.

**Pull request closed without merging**